### PR TITLE
Disable name prefix matching when updating/deleting entities

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/client.py [hunk 1]
```diff
@@ -3822,23 +3880,29 @@ def _get_entity_by_prefix(
...
-            logical_operator=LogicalOperators.OR,
-            name=f"contains:{partial_id_or_name}",
-            id=f"startswith:{partial_id_or_name}",
-        )
+        list_method_args: Dict[str, Any] = {
+            "logical_operator": LogicalOperators.OR,
+            "id": f"startswith:{partial_id_or_name}",
+        }
+        if allow_name_prefix_match:
+            list_method_args["name"] = f"startswith:{partial_id_or_name}"
```
Comment (stefannica) [inline src/zenml/client.py:3888]:
```suggestion
        if allow_name_prefix_match:
            list_method_args["name"] = f"startswith:{partial_id_or_name}"
        else:
            list_method_args["name"] = partial_id_or_name        
```

I think you missed this

File: tests/integration/functional/test_client.py [hunk 2]
```diff
@@ -964,3 +965,134 @@ def test_deleting_deployments(clean_client):
...
+    get_args: Dict[str, Any] = {}
+    update_args: Dict[str, Any] = {}
+    delete_args: Dict[str, Any] = {}
+
+
+crud_test_configs = [
+    ClientCrudTestConfig(
+        entity_name="user",
+        create_args={"name": "user_name"},
+        update_args={"updated_name": "updated_user_name"},
```
Comment (stefannica) [inline tests/integration/functional/test_client.py:982]: I love this generalized way of testing the client !

Could you please use some way to randomize the names of the created resources, similar to what we do in the zen_store generic CRUD tests ? As these tests will also be run on shared CI environments, they will trample over one-another. Using random names also makes it easier to run these tests in parallel in the future, or in another dimension.

File: tests/integration/functional/test_client.py [hunk 3]
```diff
@@ -964,3 +965,134 @@ def test_deleting_deployments(clean_client):
...
+    """Tests basic CRUD method on the client."""
+    create_method = getattr(
+        clean_client, f"create_{crud_test_config.entity_name}"
+    )
+    get_method = getattr(clean_client, f"get_{crud_test_config.entity_name}")
+    delete_method = getattr(
+        clean_client, f"delete_{crud_test_config.entity_name}"
+    )
+
+    entity = create_method(**crud_test_config.create_args)
```
Comment (stefannica) [inline tests/integration/functional/test_client.py:1052]: Could you please wrap this into a nice try/finally block and clean up any remaining resources at the end using the generated random name? If the test fails mid-way, those will be left on the test environment.

**Review by stefannica — CHANGES_REQUESTED** [review]
Thanks for making this a better UX, I know it couldn't have been easy going through all those client API functions.

File: src/zenml/client.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/client.py:3888]: As far as I understood, that case is already handled in the method that calls this, which covers both the exact ID and exact name match

Comment (stefannica) [reply src/zenml/client.py:3888]: Ah, of course. You're right.

**Review by stefannica — COMMENTED** [review]
LGTM

**Pull request merged**